src/Features/Core/Portable/Diagnostics/DiagnosticService_UpdateSourceRegistrationService.cs

```diff
@@ -31,5 +32,23 @@ public void Register(IDiagnosticUpdateSource source)
                 source.DiagnosticsUpdated += OnDiagnosticsUpdated;
             }
         }
+
+        public void Shutdown(IDiagnosticUpdateSource source, Workspace workspace)
+        {
+            var eventToken = _listener.BeginAsyncOperation(nameof(Shutdown));
+            _eventQueue.ScheduleTask(() =>
```

Comment:
Can you add a comment explaining why this is being done in a deferred way?

Reply:
sure.